projects/plugins/jetpack/sal/class.json-api-post-base.php

```diff
@@ -1,74 +1,208 @@
-<?php
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
 /**
  * This class wraps a WP_Post and proxies any undefined attributes
  * and methods to the wrapped class. We need to do this because at present
  * the WP_Post class is marked as final (in 4.5 this will change, though it's
  * not clear if there will be a mechanism to retrieve from the DB into the over-
  * ridden class dynamically).
- **/
+ *
+ * @package automattic/jetpack
+ */
 
-require_once dirname( __FILE__ ) . '/class.json-api-metadata.php';
-require_once dirname( __FILE__ ) . '/class.json-api-date.php';
+require_once __DIR__ . '/class.json-api-metadata.php';
+require_once __DIR__ . '/class.json-api-date.php';
 require_once ABSPATH . 'wp-admin/includes/post.php';
 require_once ABSPATH . 'wp-includes/post.php';
 
+/**
+ * Base class for SAL_Post.
+ */
 abstract class SAL_Post {
+
+	/**
+	 * A WP_Post instance.
+	 *
+	 * @var WP_Post
+	 */
 	public $post;
+
+	/**
+	 * The post request context (for example 'edit' or 'display')
+	 *
+	 * @var string
+	 */
 	public $context;
+
+	/**
+	 * A Jetpack_Site instance.
+	 *
+	 * @var Jetpack_Site
+	 */
 	public $site;
 
-	function __construct( $site, $post, $context ) {
-		$this->post = $post;
+	/**
+	 * Constructor function
+	 *
+	 * @param Jetpack_Site $site A Jetpack_Site instance.
+	 * @param WP_Post      $post A WP_Post instance.
+	 * @param string       $context The post request context (for example 'edit' or 'display').
+	 */
+	public function __construct( $site, $post, $context ) {
+		$this->post    = $post;
 		$this->context = $context;
-		$this->site = $site;
+		$this->site    = $site;
 	}
 
+	/**
+	 * Setting this WP_Post instance's key value
```

Comment:
`__set`, `__get`, `__call`, and `__isset` are all "[magic methods](https://www.php.net/manual/en/language.oop5.magic.php)". The doc comments can probably assume the reader knows that and focus on the fact that these are all forwarding to the wrapped WP_Post object.